File: Python_Programs/test/test_Program_5.py
import unittest
from src.Program_5 import concatenate_array_twice
from utils.test_utils import print_test_case


class TestProgram5(unittest.TestCase):

    def test_rearrange_array(self):
        test_cases = [
            (1, [1, 2, 1], [1, 2, 1, 1, 2, 1]),
            (2, [1, 3, 2, 1], [1, 3, 2, 1, 1, 3, 2, 1])
        ]

        for num, input_array, expected_output in test_cases:
            actual_output = concatenate_array_twice(input_array)
            print_test_case(num, input_array, expected_output, actual_output)
            self.assertEqual(actual_output, expected_output)


if __name__ == "__main__":
    unittest.main()

File: Java_Programs/test/com/example/tests/Test_Program_5.java
package com.example.tests;

import com.example.programs.Program_5;
import com.example.utils.TestUtils;
import org.junit.Test;
import static org.junit.Assert.*;

public class Test_Program_5 {
    @Test
    public void testConcatenateArrayTwice() {
        // Define test cases
        Object[][] testCases = {
                {1, new int[]{1, 2, 1}, new int[]{1, 2, 1, 1, 2, 1}},
                {2, new int[]{1, 3, 2, 1}, new int[]{1, 3, 2, 1, 1, 3, 2, 1}}
        };

        for (Object[] testCase : testCases) {
            int num = (int) testCase[0];
            int[] input = (int[]) testCase[1];
            int[] expectedOutput = (int[]) testCase[2];
            int[] actualOutput = Program_5.concatenateArrayTwice(input);

            // Print test case results
            TestUtils.printTestCase(num, input, expectedOutput, actualOutput);

            // Assert the result
            assertArrayEquals(expectedOutput, actualOutput);
        }
    }
}
